### ISSUE TITLE
Change hoverability after initialization

### ISSUE THREAD

**Issue by sanderdv** [NickDJM/accessible-menu#84]
Hi Nick,
First of all, thanks for the great work! I'm trying to change the `isHoverable` setting _after_  initialization, depending on screen size, because my menu changes to a hamburger menu below a certain screen width. At that point I would like to disable hoverability.

I setup a `DisclosureMenu` like this:
```js
let mainMenu = new AccessibleMenu.DisclosureMenu({
      menuElement: document.querySelector('ul.nav'),
      menuItemSelector: '.menu-item',
      menuLinkSelector: '.menu-item > a',
      submenuItemSelector: '.menu-item-has-children',
      submenuToggleSelector: '.menu-item-has-children > a',
      submenuSelector: '.sub-menu',
      openClass: 'open',
      isHoverable: true,
});
```
And I change the `isHoverable` setting to `false` when the window is below a certain width (using jQuery):
```js
$(window).on('load resize', function(){
    if($(window).width() <= 860) {
        mainMenu.isHoverable = false;
    }
});
```
When I console.log the mainMenu after that, I can see that the `hoverable` setting is indeed `false`, but the `mouseenter` and `mouseleave` events are still working.

I tried to create a new `DisclosureMenu`, but the events are still there from the 'old' one.
I tried to run the `initialize()` method on the `mainMenu` after changing `isHoverable()`.

I think that there is currently no way to remove the event listeners on the submenu toggles, unless I do it manually. Or to destroy (and by that I mean destroy all event listeners) a `DisclosureMenu`. I hope you understand what I'm trying to do and hopefully can point me in the right direction.

**Comment by NickDJM** [comment]
Hey @sanderdv, this is definitely an interesting dilemma- resizing a window would definitely cause some issues when it comes to the `isHoverable` property...

I believe I never ran into this issue because I only tested collapsed menus on mobile screens (which don't trigger hover events at all), but with a mouse this definitely would be an issue.

We could add a quick if statement in the hover event listeners to make sure the menu is indeed hoverable before it opens/closes the menu... However you'd still need to re-run `initialize()` for it to update all sub-menus.

Ultimately I'd like to see this automatically update all sub-menus whenever _any_ property is updated, but that will require a pretty big rewrite... Reactivity is definitely on my mind for v3.0.0!

**Comment by NickDJM** [comment]
You can check out a possible fix for this issue in the attached PR.

I'll see if I can do a few tests to make sure it doesn't have any unintended side effects when I have a bit of time to do so.

**Comment by sanderdv** [comment]
Hi @NickDJM ,
Thanks for the response and fix! It worked, but also brought up some other issue (I commented on it in the PR).

I realize that my case is pretty uncommon. After some more thought, I think I really shouldn't change hoverability based on screen size. I should check if a device is able to have actual hover events, which won't change after initialization.

I still think it would be pretty cool to be able to update properties after initialization, and there will probably be valid use cases for it. So I'm looking forward to a v3.0.0!

**Comment by NickDJM** [comment]
@sanderdv Yeah I think the re-initialization is what's causing the weirdness to occur, I think I'm going to postpone this issue until I can get some reactivity into the menu, and that should resolve this problem entirely.

**Comment by NickDJM** [comment]
Removing the v3 flag for this particular issue.

I've done some testing and while you'd _still_ need to re-initialize to automatically update all submenus, you can actually just loop through them yourself and change them.

```js
  menu.elements.submenuToggles.forEach(toggle => {
    toggle.elements.controlledMenu.isHoverable = !menu.isHoverable;
  });

  menu.isHoverable = !menu.isHoverable;
```

Something like the above snippet changes all menus pretty easily and I've done a bunch of testing with toggling them on/off multiple times and everything looks like its working.

The PR mentioned in this issue has been merged in and will be in the next release.

**Comment by NickDJM** [comment]
This is now released in v2.2.0
